tests/rest/common.go

```diff
@@ -0,0 +1,100 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/verdverm/frisby"
+)
+
+// common constants used by REST API tests
+const (
+	apiURL              = "http://localhost:8080/api/v1/"
+	contentTypeHeader   = "Content-Type"
+	contentLengthHeader = "Content-Length"
+
+	authHeaderName = "x-rh-identity"
+
+	// ContentTypeJSON represents MIME type for JSON format
+	ContentTypeJSON = "application/json; charset=utf-8"
+
+	// ContentTypeText represents MIME type for plain text format
+	ContentTypeText = "text/plain; charset=utf-8"
+)
+
+// StatusOnlyResponse represents response containing just a status
+type StatusOnlyResponse struct {
+	Status string `json:"status"`
+}
+
+// setAuthHeaderForOrganization set authorization header to request
+func setAuthHeaderForOrganization(f *frisby.Frisby, orgID int) {
+	plainHeader := fmt.Sprintf("{\"identity\": {\"internal\": {\"org_id\": \"%d\"}}}", orgID)
+	encodedHeader := base64.StdEncoding.EncodeToString([]byte(plainHeader))
+	f.SetHeader(authHeaderName, encodedHeader)
+}
+
+// setAuthHeader set authorization header to request for organization 1
+func setAuthHeader(f *frisby.Frisby) {
+	setAuthHeaderForOrganization(f, 1)
+}
+
+// readStatusFromResponse reads and parses status from response body
+func readStatusFromResponse(f *frisby.Frisby) StatusOnlyResponse {
+	response := StatusOnlyResponse{}
+	text, err := f.Resp.Content()
+	if err != nil {
+		f.AddError(err.Error())
+	} else {
+		err := json.Unmarshal(text, &response)
+		if err != nil {
+			f.AddError(err.Error())
+		}
+	}
+	return response
+}
+
+// sendAndExpectStatus sends the request to the server and checks whether expected HTTP code (status) is returned
+func sendAndExpectStatus(f *frisby.Frisby, expectedStatus int) {
+	f.Send()
+	f.ExpectStatus(expectedStatus)
+	f.PrintReport()
+}
+
+// checkGetEndpointByOtherMethods checks whether a 'GET' endpoint respond correctly if other HTTP methods are used
+func checkGetEndpointByOtherMethods(endpoint string) {
+	f := frisby.Create("Check the end point " + endpoint + " with wrong method: POST").Post(apiURL)
+	sendAndExpectStatus(f, 405)
+
+	f = frisby.Create("Check the entry point " + endpoint + " with wrong method: PUT").Put(apiURL)
+	sendAndExpectStatus(f, 405)
+
+	f = frisby.Create("Check the entry point " + endpoint + " with wrong method: DELETE").Delete(apiURL)
+	sendAndExpectStatus(f, 405)
+
+	f = frisby.Create("Check the entry point " + endpoint + " with wrong method: PATCH").Patch(apiURL)
+	sendAndExpectStatus(f, 405)
+
+	f = frisby.Create("Check the entry point " + endpoint + " with wrong method: OPTIONS").Options(apiURL)
+	sendAndExpectStatus(f, 405)
+
+	f = frisby.Create("Check the entry point " + endpoint + " with wrong method: HEAD").Head(apiURL)
+	sendAndExpectStatus(f, 405)
```

Comment:
Is it testing against the `endpoint`? If I understand this code correctly, it is always testing against `apiURL`, which is the "base API endpoint". Isn't it?

Reply:
oh my fail. Fixed, and TYVM